packages/jsii-pacmak/lib/targets/go/package.ts

```diff
@@ -253,17 +254,38 @@ export class RootPackage extends Package {
     code.line();
     code.open('require (');
     // Strip " (build abcdef)" from the jsii version
-    code.line(
-      `${JSII_RT_MODULE_NAME} v${this.assembly.jsiiVersion.replace(
-        / .*$/,
-        '',
-      )}`,
-    );
-    for (const dep of this.packageDependencies) {
+    code.line(`${JSII_RT_MODULE_NAME} v${VERSION}`);
```

Comment:
The previous code was actually incorrect - the version of the jsii runtime to link against must correspond to the version of `jsii-pacmak` that generated the code, NOT the version of `jsii` that created the assembly.